File: src/test/java/kitchenpos/menu/tobe/domain/MenuPriceTest.java
package kitchenpos.menu.tobe.domain;

import kitchenpos.exception.IllegalPriceException;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.NullSource;
import org.junit.jupiter.params.provider.ValueSource;

import static org.junit.jupiter.api.Assertions.assertThrows;

class MenuPriceTest {

    @ParameterizedTest
    @NullSource
    @DisplayName("금액 정보는 필수로 입력해야한다.")
    void priceFail1(final Long input) {

        Assertions.assertThatExceptionOfType(IllegalPriceException.class)
                .isThrownBy(() -> new MenuPrice(input));
    }

    @ParameterizedTest
    @ValueSource(longs = {-1L, -10_000L})
    @DisplayName("0원보다 적은 금액을 입력하는 경우 메뉴가격을 등록할 수 없다.")
    void priceFail1(final long input) {

        assertThrows(IllegalPriceException.class, () -> new MenuPrice(input));
    }
}

File: src/test/kotlin/kitchenpos/menu/tobe/domain/MenuPriceTest.kt
package kitchenpos.menu.tobe.domain

import java.math.BigDecimal
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class MenuPriceTest {

    @Test
    @DisplayName("MenuPrice를 생성한다")
    fun createMenuPrice() {
        val price = BigDecimal.valueOf(1000L)

        val menuPrice = MenuPrice(price)

        assertEquals(price, menuPrice.price)
    }

    @Test
    @DisplayName("MenuPrice는 0원 이상이어야 한다")
    fun createMenuPriceFail() {
        assertThrows(IllegalArgumentException::class.java) {
            MenuPrice(BigDecimal.valueOf(-1))
        }
    }
}
